Portfolio: Added contact form with validation

gabrielyea opened the pull request:
## Pull request to merge contact form on main branch :mag: 
- Added input fields with validation and correct data types.
- Added formspree action link.
- Project was done collaborative with the help of my programming partner @miguelesco :+1: 
- Update README with new screenshots. :camera:

miguelesco committed:
float exercise

miguelesco committed:
merge

miguelesco committed:
Add: Contact form and action form

gabrielyea committed:
Merge Fix

gabrielyea committed:
Add Validation To Fields And Set Action To Formspree

gabrielyea committed:
Add Styling To Valid And Invalid Cases

gabrielyea committed:
Add contact-form screenshot

gabrielyea committed:
Update README.md

MahmoudBakr23 commented on index.html:
```diff
@@ -264,7 +264,52 @@ <h4 class="dark"> <span class="figure-3"></span> Skills </h4>
     </div>
 
     <div class="card">
+      <div class="card-contact-me">
+        <section>
```
- [x] **OPTIONAL** Please keep in mind that the container of the section should be a `<section>` tag, and not a `<div>` tag first!
According to the [Semantic HTML](https://www.w3schools.com/html/html5_semantic_elements.asp), it'd be way too better for the website reading and optimizing.

Davidosky007 replied:
necessary changes were made 👍

MahmoudBakr23 commented on index.html:
```diff
@@ -264,7 +264,52 @@ <h4 class="dark"> <span class="figure-3"></span> Skills </h4>
+              so if you'd like to chat please get in touch.
+            </h3>
+          </header>
+
```
- [x] Kindly remove this blank line here.
Please notice that blank lines are only added for good reasons such as separating two logical pieces of code or two large sections apart from each other. Otherwise, we make sure to delete them in order to maintain the optimization of the website and the code clean.

Davidosky007 replied:
necessary changes were made

MahmoudBakr23 commented on style.css:
```diff
@@ -395,6 +401,20 @@ footer .contact-info ul {
   width: 100%;
 }
 
+fieldset {
+  border: 0;
+}
+
+input:focus:invalid {
+  box-shadow: 0 0 5px 1px red;
+  outline: none;
+}
+
+input:focus:valid {
+  box-shadow: 0 0 5px 1px green;
+  outline: none;
+}
+
```
- [x] Please make sure to follow the given design for this section. It helps to make the website more professional, consistent, and good-looking.
Also, it's a good practice for you in order to get used to following design template guidelines.

> Keep in mind that you should follow the given design in typography, font size, font weight, colors, images, spacings (margin and padding), and button interactions such as hover, active (when it's pressed), disabled, and enable (default state).

_Kindly take a look at the images down below, then do the corresponding changes._

### **Your design**
![Screenshot1](https://user-images.githubusercontent.com/47625407/121253742-c4b29c00-c8a9-11eb-901f-6e48bec5632c.png)
### **[Original design](https://www.figma.com/file/l7SqJ3ZfkAKih9sFxvWSR4/Microverse-Student-Project-1?node-id=48%3A27)**
![Screenshot](https://user-images.githubusercontent.com/47625407/121253798-d5631200-c8a9-11eb-9f9b-06b94a29f7ac.png)

gabrielyea replied:
I wasn't sure if the red outline was ok just for giving feedback on invalid data, will remove it!

MahmoudBakr23 replied:
The red outline was showing when I select the input field. Also, you've got the footer and the title of the section unstyled the same as the design template.

_Please reply here in case you have any questions._
Thank you for replying.

MahmoudBakr23 reviewed (changes requested):
Hello Team @gabrielyea @miguelesco 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Linter is correct. 💯
- Descriptive README, PR summary, and commit messages. 💯
- Correct HTML form validations. 💯 
- The Live Demo Link 💯 

However, there is always room for improvement.

> Keep in mind that your long hours of work are appreciated. However, the code review process is for improving the code you wrote, and not for criticizing it. 😄

 ## Required Changes ♻️

_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

_Please check the **OPTIONAL** comments under the review._

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

gabrielyea committed:
Change div for section and Add correct style to footer

gabrielyea committed:
Add correct font to contact-me header

gabrielyea committed:
Merge branch 'contact-form' of github.com:gabrielyea/module1-portfolio-setup into contact-form

gabrielyea committed:
Fix linter errors

gabrielyea commented:
## New push! :rocket: 
- Replaced divs with sections as requested.
- Deleted white space (if I missed some I'm sorry linter doesn't show them :disappointed: )
- Added missing styling on footer and contact title.

Davidosky007 commented on index.html:
```diff
@@ -202,71 +203,82 @@ <h3>Profesional Art Printing Data</h3>
+    </header>
+    <form action="https://formspree.io/f/mbjqyrqj" method="POST">
+      <fieldset>
+        <input oninvalid="setCustomValidity('Please enter your full name')" type="text" name="name" id="fullname"
+          required maxlength="30" placeholder="Full name">
+        <input type="email" name="_replyto" oninvalid="setCustomValidity('Please enter a valid email')" id="email"
+          required placeholder="Email address">
```
The form is not submitting even when all fields are properly filled please kindly fix this  this is what it looks like 
![Screenshot (21)](https://user-images.githubusercontent.com/52497246/121265509-d3508180-c8b0-11eb-83d2-d6c79a3767f3.png)

MahmoudBakr23 replied:
That's odd
it's working just fine with me 
![Screenshot_2021-06-09-00-32-27-18](https://user-images.githubusercontent.com/47625407/121266139-4ad6de80-c8ba-11eb-83e9-9a2d3dc48db5.jpg)
![Screenshot_2021-06-09-00-32-18-69](https://user-images.githubusercontent.com/47625407/121266146-4ca0a200-c8ba-11eb-8502-5cc47cd09ddc.jpg)

gabrielyea replied:
Yes it was also working fine with me but just now that I checked it is no longer working

MahmoudBakr23 replied:
It's still valid for me even now.

gabrielyea replied:
Now its working again 
![image](https://user-images.githubusercontent.com/28903404/121266551-4ab9ed00-c880-11eb-8f9c-0ae2f2495315.png)
![image](https://user-images.githubusercontent.com/28903404/121266564-50173780-c880-11eb-8ed8-4841127757ed.png)

gabrielyea replied:
What should I do? :fearful:

MahmoudBakr23 replied:
https://formspree.io/f/myylerlb

Try this one instead of yours and test it locally, then let me know here please

MahmoudBakr23 replied:
It works 👍🏻

gabrielyea replied:
> https://formspree.io/f/myylerlb
> 
> Try this one instead of yours and test it locally, then let me know here please

Done, I think it went thru no errors on my machine

gabrielyea replied:
> It works 👍🏻

should I merge? or is there a way to know what caused that?

MahmoudBakr23 replied:
I guess it stopped working for you on your link because you spammed emailing while you're testing it. Well, now we know you have nothing to worry about and you're good to go.
submit another CRR now. 
good luck and happy coding with the next.

gabrielyea replied:
Should I submit it with the link to my formspree or with yours?

MahmoudBakr23 replied:
it doesn't matter, but use yours. Anyways, the next reviewer will understand the situation from this conversation.

gabrielyea replied:
> it doesn't matter, but use yours. Anyways, the next reviewer will understand the situation from this conversation.

hope he does, its my last review :fearful:

Dipeshtwis replied:
It's working for me. You're good to go.

Davidosky007 replied:
It works on my end now 👍

Davidosky007 reviewed (changes requested):
Hi @gabrielyea @miguelesco, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

gabrielyea commented:
## New Review Request
- No changes were made, apparently, it works now, maybe spamming the link caused the error.

Dipeshtwis reviewed (approved):
# Status: Approved ✔️

## Hi @gabrielyea, 

Your formspree is working and submitting my info. Well Done.
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

**You may close this Pull Request and merge the changes**

  
**Keep being awesome devs and good luck with your next milestone!**
---------

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
_**Dipesh Kumar**_

--------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
